### PR TITLE
Feat. Cross border

### PR DESCRIPTION
**Pull request by sleipnir**
This PR aims to add support for communication via Nats broker when there is a need for actors to call other resident actors from other ActorSystem's

### CODE REVIEW

File: lib/actors.ex [hunk 1]
```diff
@@ -176,24 +180,90 @@ defmodule Actors do
...
-        %InvocationRequest{} = request,
+        %InvocationRequest{
+          system: %ActorSystem{name: system_name} = _system
+        } = request,
         opts \\ []
       ) do
-    invoke_with_span(request, opts)
+    case Config.get(Actors, :actor_system_name) do
+      name when name === system_name ->
+        IO.inspect({system_name, opts}, label: "Caiu aqui invoke_with_span")
```
Comment (eliasdarruda) [inline lib/actors.ex:190]:
```suggestion
```

File: lib/actors.ex [hunk 2]
```diff
@@ -176,24 +180,90 @@ defmodule Actors do
...
+        invoke_with_span(request, opts)
+
+      _ ->
+        IO.inspect({system_name, opts}, label: "Caiu aqui invoke_with_nats")
```
Comment (eliasdarruda) [inline lib/actors.ex:194]:
```suggestion
```

File: lib/spawn/utils/nats.ex [hunk 3]
```diff
@@ -0,0 +1,55 @@
+defmodule Spawn.Utils.Nats do
+  @moduledoc false
+
+  alias alias Eigr.Functions.Protocol.InvocationRequest
```
Comment (eliasdarruda) [inline lib/spawn/utils/nats.ex:4]:
```suggestion
  alias Eigr.Functions.Protocol.InvocationRequest
```

File: lib/spawn/utils/nats.ex [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+    # TODO: Get other parameters here to build complex connections
+
+    hosts_conn_map
+  end
+
+  def request(system, payload, opts \\ []) do
+    async? = Keyword.get(opts, :async, false)
+    conn = connection_name()
+    topic = "spawn.#{system}.actors.actions"
```
Comment (eliasdarruda) [inline lib/spawn/utils/nats.ex:31]: :information_source: You could move this to a function in this module and reuse it in the supervisor

**Review by eliasdarruda — APPROVED** [review]

File: lib/spawn/cluster/node/server_supervisor.ex [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+      {Gnat.ConsumerSupervisor, connection_params}
+    ]
+
+    Supervisor.init(children, strategy: :one_for_one)
+  end
+
+  defp connection_settings(name, config) do
+    %{
+      name: name,
+      backoff_period: 3_000,
```
Comment (eliasdarruda) [inline lib/spawn/cluster/node/server_supervisor.ex:52]: make this configurable?

Comment (sleipnir) [reply lib/spawn/cluster/node/server_supervisor.ex]: For now I only extract to one attribute. We can iterate this better later.

**Pull request merged**